src/modules/sensors/sensors.cpp

```diff
@@ -398,9 +421,55 @@ Sensors::diff_pres_poll(struct sensor_combined_s &raw)
 
 		_airspeed.air_temperature_celsius = air_temperature_celsius;
 		_airspeed.differential_pressure_filtered_pa = _diff_pres.differential_pressure_filtered_pa;
+	}
+
+	// update wind and airspeed estimator
+	_wind_estimator.update(raw.gyro_integral_dt);
+
+	bool fuse_airspeed = updated && hrt_elapsed_time(&_control_state.timestamp) < 1e6
+			     && (hrt_elapsed_time(&_time_last_beta_fused) > 5e4);
+	bool fuse_beta = updated && hrt_elapsed_time(&_control_state.timestamp) < 1e6
+			 && (hrt_elapsed_time(&_time_last_airspeed_fused) > 5e4);
+
+	if (fuse_beta || fuse_airspeed) {
+		matrix::Dcmf R_to_earth(matrix::Quatf(_control_state.q));
+		matrix::Vector3f vI(_control_state.x_vel, _control_state.y_vel, _control_state.z_vel);
+		vI = R_to_earth * vI;
+
+		if (fuse_beta) {
+			_wind_estimator.fuse_beta(&vI._data[0][0], _control_state.q);
+			_time_last_beta_fused = hrt_absolute_time();
+		}
+
+		if (fuse_airspeed) {
+			matrix::Vector3f vel_var(_control_state.vel_variance);
+			vel_var = R_to_earth * vel_var;
+			_wind_estimator.fuse_airspeed(_airspeed.indicated_airspeed_m_s, &vI._data[0][0], &vel_var._data[0][0]);
```

Comment:
@dagar @priseborough I used indicated airspeed here because "true airspeed" seems to be rubbish in SITL, very small values.